src/dmifn.rs

```diff
@@ -909,31 +909,17 @@ pub fn dmi_memory_module_error(error_status: u8) {
         false => println!("{}", print),
     }
 }
-pub fn dmi_cache_size(attr: &str, size1_opt: Option<u16>, size2_opt: Option<u32>) {
-    let large_opt = match (size1_opt, size2_opt) {
-        (Some(installed_size), None) => {
-            // High bit 15 is granularity.  Make it bit 31 to match installed_cache_size_2):
-            // 0 == 1K
-            // 1 == 64K
-            let size_32 = installed_size as u32;
-            Some((size_32 & 0x8000u32 << 16) | (size_32 & 0x7FFFu32))
-        }
-        (Some(_), Some(installed_size_2)) => Some(installed_size_2),
-        _ => None,
+pub fn dmi_cache_size(attr: &str, size1_opt: Option<CacheMemorySize>, size2_opt: Option<CacheMemorySize>) {
+    let size = match (size1_opt, size2_opt) {
+        (Some(CacheMemorySize::Kilobytes(size)), _) => size,
+        // Preserve behavior of dmidecode if cache size is recorded as 0xFFFF
+        // and dmi table does not include cache size 2
+        (Some(CacheMemorySize::SeeCacheSize2), None) => 0x7FFF * 64,
```

Comment:
Not actually identical to dmidecode here

Reply:
Nevermind, I confused myself because I wasn't on the correct branch, so I wasn't printing the right thing. But that actually makes me realize that in the spec, 0xFFFF is _not_ actually considered a sentinel.

Reply:
Specifically, when the SMBios version is less than 3.1.0

Reply:
We can remove this `(Some, None)` case if we merge https://github.com/jrgerber/smbios-lib/pull/101 But it also might not make sense to support older spec versions in smbios-lib